provider/datagen/src/lib.rs

```diff
@@ -246,18 +235,21 @@ pub fn keys_from_bin<P: AsRef<Path>>(path: P) -> std::io::Result<Vec<DataKey>> {
         } else if file[i..].starts_with(icu_provider::trailing_tag!().as_bytes())
             && last_start.is_some()
         {
-            candidates.insert(&file[last_start.unwrap()..i]);
+            if let Some(key) = std::str::from_utf8(&file[last_start.unwrap()..i])
+                .ok()
+                .and_then(key)
```

Comment:
Nit: This code is not great for readability; you're using `key` as a function here because there happens to be a function with that name in scope, but you are also setting the local variable named `key` in the same expression.